internal/hcsoci/resources_lcow.go

```diff
@@ -94,6 +94,9 @@ func allocateLinuxResources(ctx context.Context, coi *createOptionsInternal, res
 				}
 				resources.scsiMounts = append(resources.scsiMounts, hostPath)
 				coi.Spec.Mounts[i].Type = "none"
+				if mount.Destination == "/tmp" {
```

Comment:
I think it might be best to use a new `mount.Type` like `autocleanup-virtual-disk`, that acts as a virtual disk with the additional semantic that it is cleaned up when the container exits.

Similarly to the other PR, if we see this change as "definitely will be removed, just needed to unblock testing", I'm fine with not trying to generalize the functionality.

Reply:
I see this as a total hack that we should be removing. The shim should not manage the lifetime of a vhd as the storage is managed by the CSI which is orchestrator level.

Reply:
Okay, then go nuts :)